File: finalysis-spider/src/main/java/xyz/liujin/finalysis/spider/entity/KLine.java
package xyz.liujin.finalysis.spider.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(autoResultMap = true)
public class KLine {
    @TableId(type = IdType.AUTO )
    private Integer id;
    private String stockCode;
    private OffsetDateTime dateTime;
    private BigDecimal open;
    private BigDecimal close;
    private BigDecimal high;
    private BigDecimal low;
    private BigDecimal inc;
    private BigDecimal incRate;
    private Integer volume;
    private BigDecimal amount;
    private BigDecimal volumeRatio;
    private BigDecimal turn;
    private BigDecimal committee;
    private BigDecimal selling;
    private BigDecimal buying;
}

File: pystock/database/entity/kline.py
from pystock.database.tool import date_tool

class KLine:
    def __init__(self, stock_code, start_time=date_tool.now(), end_time=date_tool.now(), 
    open=0.00, close=0.00, high=0.00, low=0.00, 
    volume=0, turnover=0.00, 
    volume_ratio=0.00, turnover_rate=0.00, committee=0.00, 
    selling=0.00, buying=0.00):
        self.stock_code = stock_code
        self.start_time = start_time
        self.end_time = end_time
        self.open = open
        self.close = close
        self.high = high
        self.low = low
        self.volume = volume
        self.turnover = turnover
        self.volume_ratio = volume_ratio
        self.turnover_rate = turnover_rate
        self.committee = committee
        self.selling = selling
        self.buying = buying
